src/ort_genai_c.cpp

```diff
@@ -370,6 +370,47 @@ OgaResult* OGA_API_CALL OgaGenerator_GetOutput(const OgaGenerator* oga_generator
   OGA_CATCH
 }
 
+OgaResult* OGA_API_CALL OgaGenerator_GetLogits(OgaGenerator* oga_generator, OgaTensor** out) {
+  OGA_TRY
+  auto logits_span = reinterpret_cast<Generators::Generator*>(oga_generator)->GetLogits();
+  auto& generator = *reinterpret_cast<const Generators::Generator*>(oga_generator);
+  const std::array<int64_t, 3> shape{generator.state_->params_->search.batch_size, 1, generator.model_->config_->model.vocab_size};
+  std::span<const float> cpu_logits_span = logits_span.CopyDeviceToCpu();
+
+  // Copy logits to cpu tensor
+  std::unique_ptr<OrtValue> ortvalue_clone = OrtValue::CreateTensor(generator.model_->allocator_cpu_,
+                                                                    shape,
+                                                                    ONNX_TENSOR_ELEMENT_DATA_TYPE_FLOAT);
```

Comment:
Thought of one more:

```suggestion
  std::unique_ptr<OrtValue> ortvalue_clone = OrtValue::CreateTensor<float>(generator.model_->allocator_cpu_, shape);
```